openforcefield/typing/engines/smirnoff/parameters.py

```diff
@@ -1372,18 +1414,36 @@ def check_handler_compatibility(self,
         #self._long_range_dispersion:'long_range_dispersion'
         #}
 
-    # TODO: nonbondedMethod and nonbondedCutoff should now be specified by StericsForce attributes
     def create_force(self, system, topology, **kwargs):
 
+        self._validate_parameters()
+
+        # If there's no cutoff, make sure the system is nonperiodic/vacuum
+        if (self._long_range_dispersion == "None") and (not topology.is_vacuum):
+            raise SMIRNOFFSpecError("If vdW long_range_dispersion is None, a vacuum/nonperiodic Topology "
+                                    "must be provided")
+        # If there is a cutoff, make sure the system is periodic/condensed
+        if (self._long_range_dispersion != "None") and (not topology.is_condensed):
+            raise SMIRNOFFSpecError("If vdW long_range_dispersion is isotropic or LJPME, a condensed/periodic Topology "
+                                    "must be provided")
+
+
         force = openmm.NonbondedForce()
-        nonbonded_method = self._NONBOND_METHOD_MAP[self._nonbonded_method]
-        force.setNonbondedMethod(nonbonded_method)
-        force.setCutoffDistance(self._cutoff.in_units_of(unit.nanometer))
-        if 'ewaldErrorTolerance' in kwargs:
-            force.setEwaldErrorTolerance(kwargs['ewaldErrorTolerance'])
-        if 'useDispersionCorrection' in kwargs:
-            force.setUseDispersionCorrection(
-                bool(kwargs['useDispersionCorrection']))
+        # If we're using PME, then the only possible openMM Nonbonded type is LJPME
+        if self._long_range_dispersion == 'LJPME':
+            force.setNonbondedMethod(openmm.NonbondedForce.LJPME)
+            force.setCutoffDistance(self._cutoff.in_units_of(unit.nanometer))
```

Comment:
The `.in_units_of(unit.nanometer)` is totally unnecessary here (and in subsequent lines) if the quantity is a `simtk.unit.Quantity`. It will already be automatically converted to the appropriate units by OpenMM.

Reply:
Ah, great. I wasn't sure why `in_units_of` was there in the first place, but I didn't want to break the pattern in case it was important. Cleaned up in fa632e2